arex-schedule-web-api/src/main/java/com/arextest/schedule/planexecution/impl/PlanExecutionMonitorImpl.java

```diff
@@ -19,70 +21,60 @@
 @Slf4j
 public class PlanExecutionMonitorImpl implements PlanExecutionMonitor {
 
-  public static final int SECOND_TO_REFRESH = 5;
   @Resource
   private ScheduledExecutorService monitorScheduler;
   @Resource
   private List<PlanMonitorHandler> planMonitorHandlerList;
 
   @Override
-  public void monitorOne(ReplayPlan task) {
-    if (task == null) {
+  public void register(ReplayPlan plan) {
+    if (CollectionUtils.isNotEmpty(plan.getMonitorFutures())) {
+      LOGGER.info("duplicate register monitor task {}", plan.getId());
       return;
     }
-    LOGGER.info("Monitoring task {}", task.getId());
-    for (PlanMonitorHandler handler : planMonitorHandlerList) {
-      try {
-        handler.handle(task);
-      } catch (Throwable t) {
-        LOGGER.error("Error handling plan:{}", task.getId(), t);
+
+    try {
+      LOGGER.info("register monitor task {}", plan.getId());
+      List<ScheduledFuture<?>> monitorFutures =
+          Lists.newArrayListWithCapacity(planMonitorHandlerList.size());
+
+      for (PlanMonitorHandler handler : planMonitorHandlerList) {
+        ScheduledFuture<?> monitorFuture = monitorScheduler
+            .scheduleAtFixedRate(new AbstractTracedRunnable() {
+                                   @Override
+                                   protected void doWithTracedRunning() {
+                                     try {
+                                       handler.handle(plan);
+                                     } catch (Exception e) {
+                                       LOGGER.error("failed to handle monitor task. plan:{}", plan.getId(), e);
+                                     }
+                                   }
+                                 }, 0, handler.period(),
+                TimeUnit.SECONDS);
+        monitorFutures.add(monitorFuture);
       }
+      plan.setMonitorFutures(monitorFutures);
+    } catch (Exception e) {
+      LOGGER.error("failed to register monitor task. plan:{}", plan.getId(), e);
     }
   }
 
-  @Override
-  public void register(ReplayPlan plan) {
-    MonitorTask task = new MonitorTask(plan);
-    ScheduledFuture<?> monitorFuture = monitorScheduler
-        .scheduleAtFixedRate(task, 0, SECOND_TO_REFRESH, TimeUnit.SECONDS);
-    plan.setMonitorFuture(monitorFuture);
-  }
-
   @Override
   public void deregister(ReplayPlan plan) {
-    plan.getMonitorFuture().cancel(false);
-    LOGGER.info("deregister monitor task, planId: {}", plan.getId());
+    List<ScheduledFuture<?>> monitorFutures = plan.getMonitorFutures();
+    if (CollectionUtils.isEmpty(monitorFutures)) {
+      return;
+    }
 
-    for (PlanMonitorHandler handler : planMonitorHandlerList) {
-      try {
+    try {
+      LOGGER.info("deregister monitor task, planId: {}", plan.getId());
+      monitorFutures.forEach(future -> future.cancel(false));
+      for (PlanMonitorHandler handler : planMonitorHandlerList) {
         handler.end(plan);
```

Comment:
Here you need to catch exceptions to ensure that other handlers execute properly.